controllers/etcd_custodian_controller.go

```diff
@@ -168,6 +384,78 @@ func (ec *EtcdCustodian) updateEtcdStatus(ctx context.Context, logger logr.Logge
 	})
 }
 
+func (ec *EtcdCustodian) reconcileJob(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd, renderedChart *chartrenderer.RenderedChart) (*batchv1.Job, error) {
+	logger.Info("Reconcile etcd compaction job")
+
+	var job batchv1.Job
+	err := ec.Get(ctx, types.NamespacedName{Name: getJobName(etcd), Namespace: etcd.Namespace}, &job)
+
+	if err == nil {
+		// Delete job and reque if the last job failed
+		if job.Status.Failed > 0 {
+			err = ec.Delete(ctx, &job, client.PropagationPolicy(metav1.DeletePropagationForeground))
+			if err != nil {
+				return nil, err
+			}
+			return nil, fmt.Errorf("job status failed")
+		}
+
+		// Delete job and reque if the job doesn't have label with this etcd instance
+		if val, ok := job.Labels["instance"]; ok {
+			if val != etcd.Name {
+				err = ec.Delete(ctx, &job, client.PropagationPolicy(metav1.DeletePropagationForeground))
+				if err != nil {
+					return nil, err
+				}
+				return nil, fmt.Errorf("job doesn't have label for this etcd instance")
+			}
+		}
+		return nil, nil
+	}
+
+	// Required job doesn't exist. Create new
+	j, err := ec.getJobFromEtcd(etcd, renderedChart, logger)
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Info("Creating job", "job", kutil.Key(j.Namespace, j.Name).String())
+	err = ec.Create(ctx, j)
```

Comment:
I think this will fail if the previous compaction was successful because nothing deleted the previous `Job` right?

Reply:
Job will be immediately deleted after successful completion. It can be mentioned in job spec